### PR TITLE
landscape.yaml split

### PR DESCRIPTION
**Pull request by Diaphteiros**
`landscape.yaml` is splitted into `landscape_base.yaml` and `user_config.yaml`.

### CODE REVIEW

File: landscape_config.yaml [hunk 1]
```diff
@@ -94,33 +94,40 @@ misc:
...
   # htpasswd -bnBC 10 "" password | tr -d ':\n' | sed 's/$2y/$2a/'
   staticPasswords:
-  # sample
-  #- email: "dirk@**********"
-  #  password: "***************"
-  #  username: "admin"
-  #  userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
+  # example 
+  - email: "some.mail@example.org"
+    hash: "$2a$10$JIKb36h2oM5pVAvvM/EniemkQJhCtdmmsVamyRmNVJhCyOlTY72uu" # example hash for password "password"
```
Comment (msohn) [inline landscape_config.yaml]: I would write this like you did elsewhere, and remove the unnecessary comments:

 email: \<email address>\>
 hash: \<bcrypted password\>
 username: \<user name\>
 userID: \<UUID\>

**Review by msohn — COMMENTED** [review]
will you also address the other comments I gave ?

File: user_config.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  master:
+    # Properties for master nodes.
+    count: 3
+    volume_size: 50
+  worker:
+    # Properties for worker nodes
+    count: 3
+    volume_size: 50
+
+misc:
```
[the remark this answers is not in the document]
Comment (Diaphteiros) [reply user_config.yaml]: I agree, but I didn't find a better name for this / a better node to group the values

File: landscape_base.yaml [hunk 3]
```diff
@@ -169,15 +169,15 @@ charts:
       internalDomain:
         # the cluster internal domain, this should be set to
         # "internal." + clusters.dns.domain_name
-        domain: "(( merge ))"
+        domain: (( "internal." clusters.dns.domain_name ))
```
[the remark this answers is not in the document]
Comment (Diaphteiros) [reply landscape_base.yaml]: It concatenates "internal." with the value of the clusters.dns.domain_name node.

File: landscape_base.yaml [hunk 4]
```diff
@@ -12,13 +12,15 @@
...
-# environment. Follow the instructions in the file to configure an
-# installation
-
 # Note: the purpose of this project is to provide a simple setup of the
 #       Gardener environment. It is not meant for productive environments
 
+# This is the base file for a spiff merge file. 
+# It contains spiff merge tags that will be replaced by values 
+# from the user_config.yaml file, as well as technical configuration, 
+# that is assumed to be independent of the user configuration.
```
[the remark this answers is not in the document]
Comment (Diaphteiros) [reply landscape_base.yaml]: Hm, I googled a bit and it seems that "independent from" is mainly used when talking about countries or people, while "independent of" is better in most other cases.

File: landscape_config.yaml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+#      clientID: <github client id or environment variable containing it>
+#      clientSecret: <github client secret or environment variable containing it>
+#      # redirectURI: <identity URL (with https!)>/callback
+#      # identity is usually located at identity.ingress. + clusters.dns.domain_name
+#      redirectURI: <redirect URI>
+#      # Only users which are members of at least one organization can authenticate
+#      orgs:
+#      - name: <name of github organization>
+- name: certmanager
+  # Email address used for ACME registration
```
Comment (msohn) [inline landscape_config.yaml]: link to https://en.wikipedia.org/wiki/Automated_Certificate_Management_Environment ?

File: landscape_config.yaml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  email: "<email address>"
+
+seed_config:
+  variant_aws:
+    # The cluster created by Kubify is registered as the seed cluster,
+    # use the same region here as in the authentication node above
+    region: "eu-west-1"
```
Comment (msohn) [inline landscape_config.yaml]: can we refer to the key aws_region defined above ?

File: landscape_config.yaml [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    # Note that ami ids differ between regions
+    image: "ami-34237c4d"
+    zones:
+    - eu-west-1a
+    - eu-west-1b
+    - eu-west-1c
+  variant_openstack:
+    # The cluster created by Kubify is registered as the seed cluster,
+    # use the same region here as in the authentication node above
+    region: "<region name>"
```
Comment (msohn) [inline landscape_config.yaml]: can we refer to the key aws_region defined above ?

File: README.md [hunk 8]
```diff
@@ -51,19 +51,13 @@ origin so you do not accidentally publish your secrets to the public template re
```
Comment (msohn) [inline README.md]: this commit can be squashed into 3c347bab0e620eeced73969743aa099f764835e6

File: landscape_config.yaml [hunk 9]
```diff
@@ -72,6 +72,12 @@ clusters:
...
 
+etcd_backup:
+  storage_type: s3
+  region: eu-west-1
+  # access_key: <access key for s3 bucket>
+  # secret_key: <secret key for s3 bucket>
+
 misc:
   variant_all:
     # use htpasswd to create password entries
```
Comment (msohn) [inline landscape_config.yaml]: submodule update should be done in the commit doing the split of the landscape.yaml file

**Review by msohn — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -32,6 +32,44 @@ git client and the Docker runtime installed.
 Follow these steps to install Gardener. Do not proceed to the next
 step in case of errors.
 
+
+## TLDR
+If you are already familiar with the installation procedure and just want a short 
+summary of the commands you have to use, here it is:
+
+```
+# clone git repo with submodules
```
Comment (msohn) [inline README.md]: remove this comment, instead start with
 \# setup

File: README.md [hunk 11]
```diff
@@ -32,6 +32,44 @@ git client and the Docker runtime installed.
...
+./deploy_kubify.sh
+
+# deploy gardener
+./deploy_gardener.sh
+
+# optional: certmanager
+cd components
+./deploy.sh certmanager
+
+
```
Comment (msohn) [inline README.md]: remove unnecessary blank lines

File: README.md [hunk 12]
```diff
@@ -32,6 +32,44 @@ git client and the Docker runtime installed.
...
+# -------------------------------------------------------------------
+
```
Comment (msohn) [inline README.md]: \# teardown

and leave out the comments

**Review by msohn — CHANGES_REQUESTED** [review]
the ./deploy_gardener.sh could print the URLs to open to accept self-signed certs and open the Gardener dashboard, this way the user doesn't need to type the URL and can open it directly

**Review by msohn — APPROVED** [review]

**Pull request merged**